java/client/src/main/java/glide/api/models/BaseTransaction.java

```diff
@@ -2761,13 +2863,41 @@ public T zinterWithScores(@NonNull KeysOrWeightedKeys keysOrWeightedKeys) {
      */
     public T zinterWithScores(
             @NonNull KeysOrWeightedKeys keysOrWeightedKeys, @NonNull Aggregate aggregate) {
-        ArgsArray commandArgs =
-                buildArgs(
-                        concatenateArrays(
-                                keysOrWeightedKeys.toArgs(),
-                                aggregate.toArgs(),
-                                new String[] {WITH_SCORES_REDIS_API}));
-        protobufTransaction.addCommands(buildCommand(ZInter, commandArgs));
+        protobufTransaction.addCommands(
+                buildCommand(
+                        ZInter,
+                        newArgsBuilder()
+                                .add(keysOrWeightedKeys.toArgs())
+                                .add(aggregate.toArgs())
+                                .add(WITH_SCORES_REDIS_API)));
+        return getThis();
+    }
+
+    /**
+     * Returns the intersection of members and their scores from sorted sets specified by the given
+     * <code>keysOrWeightedKeys</code>.
+     *
+     * @since Redis 6.2 and above.
+     * @see <a href="https://redis.io/commands/zinter/">redis.io</a> for more details.
+     * @param keysOrWeightedKeys The keys of the sorted sets with possible formats:
+     *     <ul>
+     *       <li>Use {@link KeyArrayBinary} for keys only.
+     *       <li>Use {@link WeightedKeysBinary} for weighted keys with score multipliers.
+     *     </ul>
+     *
+     * @param aggregate Specifies the aggregation strategy to apply when combining the scores of
+     *     elements.
+     * @return Command Response - The resulting sorted set from the intersection.
+     */
+    public T zinterWithScores(
+            @NonNull KeysOrWeightedKeysBinary keysOrWeightedKeys, @NonNull Aggregate aggregate) {
```

Comment:
You can make `KeysOrWeightedKeys` parametrized with the same conditions described above.

Reply:
Will make it in another PR, I am trying to complete the entire TXN commands today, this seems like a more low priority issue ;)